### PR TITLE
feat: AP-245 AP-248 Clicking Glyph in Chat to Show Pack

### PR DESCRIPTION
**Pull request by josephmcg**
**What this PR does** 📖
- generic mobile modal that only goes partway up the screen
- desktop modal for glyph pack details on click
- mobile modal for glyph pack details on click
- send user to a specific glyph pack in marketplace on button click

**Which issue(s) this PR fixes** 🔨

Fixes #AP245-248 subissues

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

File: components/tailored/glyphs/Glyphs.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  <div class="heading">
+    <TypographyTitle :size="5" :text="selectedPack.name" />
+  </div>
+  <TailoredGlyphsSample :glyphs="samplePack" />
+  <TypographyText :text="selectedPack.description" />
+  <InteractablesButton
+    class="button"
+    type="primary"
+    size="small"
+    text="View Glyph Pack"
```
Comment (WanderingHogan) [inline components/tailored/glyphs/Glyphs.html:12]: add to translations file and reference here

Comment (josephmcg) [reply components/tailored/glyphs/Glyphs.html]: fixed

**Review by stavares843 — APPROVED** [review]
the modal on mobile is not responsive, is it not to be handled in another ticket?

otherwise, good to go

**Review by Jekrimo — APPROVED** [review]

**Comment by josephmcg** [comment]
I put in quite a bit of effort to make the modal responsive, could you clarify what you mean? The glyph details modals are responsive. The marketplace is not, that will be handled later

**Comment by stavares843** [comment]
correct branch

https://user-images.githubusercontent.com/29093946/146035786-d137cb1b-c375-40c2-9520-2dc6e9a98a02.mp4

but on mobile appears like this

Captura de ecrã 2021-12-14, às 16 09 21


when i meant the modal i meant when clicking on the glyphs section

**Comment by josephmcg** [comment]
That's not a feature that I worked on in this ticket. This was focused on clicking a glyph in a conservation after it has already been sent.

**Comment by josephmcg** [comment]
@stavares843 I fixed that issue you found, also rebased off dev and fixed. Let me know if you find any issues

**Comment by stavares843** [comment]
ohhh lol thanks for clarifying! and thanks for fixing as well 🎉

**Comment by josephmcg** [comment]
I didn't merge, just fixed conflict w/ github gui editor

**Comment by stavares843** [comment]
neat, thanks! 🎉

**Review by stavares843 — CHANGES_REQUESTED** [review]
going to a chat > glyphs > crash

Captura de ecrã 2021-12-16, às 01 38 31

**Comment by josephmcg** [comment]
thanks Sara, I must have fixed that conflict incorrectly. It should be working now

**Comment by josephmcg** [comment]
@stavares843 
ok ready for testing.

File: components/views/chat/message/glyph/Glyph.vue [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<template src="./Glyph.html" />
```
Comment (WanderingHogan) [inline components/views/chat/message/glyph/Glyph.vue:1]: can you add a closing tag to template

File: components/views/chat/message/glyph/Glyph.vue [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+<script lang="ts">
+import Vue from 'vue'
+import { mapState } from 'vuex'
+
+export default Vue.extend({
+  props: {
+    source: {
+      type: String,
+      default: '404.png',
```
Comment (WanderingHogan) [inline components/views/chat/message/glyph/Glyph.vue:10]: what is 404.png?

File: components/views/glyphs/Glyphs.vue [hunk 4]
```diff
@@ -0,0 +1,50 @@
+<template src="./Glyphs.html" />
```
Comment (WanderingHogan) [inline components/views/glyphs/Glyphs.vue:1]: can you close this template tag

**Review by WanderingHogan — APPROVED** [review]

File: components/views/chat/message/glyph/Glyph.vue (hunk 3 above)
Comment (josephmcg) [reply components/views/chat/message/glyph/Glyph.vue]: Seems to be a dummy value, I copied from another ui component. removed in favor of ''

**Review by stavares843 — COMMENTED** [review]
when you send a glyph on the chat > click on the glyph > shows the pack > click on the button to go to marketplace > opens marketplace > you cant close the marketplace - non reproducible

Captura de ecrã 2021-12-16, às 15 16 38


Chrome, macOS

**Review by stavares843 — CHANGES_REQUESTED** [review]
some glyphs have a wider button than others

Captura de ecrã 2021-12-16, às 15 18 37
Captura de ecrã 2021-12-16, às 15 18 41

**Comment by stavares843** [comment]
besides that, pretty neat! 🎉

**Comment by josephmcg** [comment]
Thanks, I'll work on those. Hogan encountered the same marketplace bug yesterday but was also unable to reproduce

**Comment by stavares843** [comment]
maybe its only on the first time going to that button 💥

**Comment by josephmcg** [comment]
Fixed the button size issue, css for modal size is much more robust for various screen sizes/open console.
Not sure about marketplace issue though

**Comment by stavares843** [comment]
thanks! :tada:

**Comment by stavares843** [comment]
neat! 🎉

**Review by stavares843 — APPROVED** [review]

**Pull request merged**